lib/ansible/modules/cloud/google/gcp_compute_disk.py

```diff
@@ -450,8 +456,35 @@ def create(module, link, kind):
     return wait_for_operation(module, auth.post(link, resource_to_request(module)))
 
 
-def update(module, link, kind):
-    module.fail_json(msg="Disk cannot be edited")
+def update(module, link, kind, fetch):
+    update_fields(module, resource_to_request(module),
+                  response_to_hash(module, fetch))
+    return fetch_resource(module, self_link(module), kind)
+
+
+def update_fields(module, request, response):
+    auth = GcpSession(module, 'compute')
+    if response.get('labels') != request.get('labels'):
+        auth.post(
+            ''.join([
+                "https://www.googleapis.com/compute/v1/",
+                "projects/{project}/zones/{zone}/disks/{name}/setLabels"
+            ]).format(**module.params),
+            {
+                u'labelFingerprint': response.get('labelFingerprint'),
+                u'labels': module.params.get('labels')
+            }
+        )
+    if response.get('sizeGb') != request.get('sizeGb'):
```

Comment:
Someday, you're going to want to inject a "not quite just literal comparisons only" bit here - a canonicalizer, we used to call them in Maps.  Don't have to do it today.

Reply:
Good call! I'm going to make a issue for this.